Assets/MixedRealityToolkit-SDK/Inspectors/UX/Interactable/InspectorBase.cs

```diff
@@ -0,0 +1,646 @@
+        {
+            // delete button
+            GUIStyle buttonStyle = new GUIStyle(GUI.skin.button);
+            
+            float buttonWidth = GUI.skin.button.CalcSize(label).x;
+
+            EditorGUILayout.BeginHorizontal();
+            GUILayout.FlexibleSpace();
+
+            bool triggered = false;
+            if (GUILayout.Button(label, buttonStyle, GUILayout.Width(buttonWidth)))
+            {
+                callback(arr, prop);
+                triggered = true;
+            }
+
+            EditorGUILayout.EndHorizontal();
+            return triggered;
+        }
+
+        public static bool FullWidthButton(GUIContent label, float padding, int index, ListButtonEvent callback, SerializedProperty prop = null)
+        {
+            GUIStyle addStyle = new GUIStyle(GUI.skin.button);
+            addStyle.fixedHeight = 25;
+            float addButtonWidth = GUI.skin.button.CalcSize(label).x * padding;
+            EditorGUILayout.BeginHorizontal();
+            GUILayout.FlexibleSpace();
+
+            bool triggered = false;
+            if (GUILayout.Button(label, addStyle, GUILayout.Width(addButtonWidth)))
+            {
+                callback(index, prop);
+                triggered = true;
+            }
+
+            GUILayout.FlexibleSpace();
+            EditorGUILayout.EndHorizontal();
+            return triggered;
+        }
+
+        public static bool FullWidthButton(GUIContent label, float padding, int[] arr, MultiListButtonEvent callback, SerializedProperty prop = null)
+        {
+            GUIStyle addStyle = new GUIStyle(GUI.skin.button);
+            addStyle.fixedHeight = 25;
+            float addButtonWidth = GUI.skin.button.CalcSize(label).x * padding;
+            EditorGUILayout.BeginHorizontal();
+            GUILayout.FlexibleSpace();
+
+            bool triggered = false;
+            if (GUILayout.Button(label, addStyle, GUILayout.Width(addButtonWidth)))
+            {
+                callback(arr, prop);
+                triggered = true;
+            }
+
+            GUILayout.FlexibleSpace();
+            EditorGUILayout.EndHorizontal();
+            return triggered;
+        }
+
+        public static bool SmallButton(GUIContent label, int index, ListButtonEvent callback, SerializedProperty prop = null)
+        {
+
+            GUIStyle smallButton = new GUIStyle(EditorStyles.miniButton);
+            float smallButtonWidth = GUI.skin.button.CalcSize(new GUIContent(label)).x;
+
+            bool triggered = false;
+            if (GUILayout.Button(label, smallButton, GUILayout.Width(smallButtonWidth)))
+            {
+                callback(index, prop);
+                triggered = true;
+            }
+            return triggered;
+        }
+
+        public static bool SmallButton(GUIContent label, int[] arr, MultiListButtonEvent callback, SerializedProperty prop = null)
```

Comment:
Kinda unclear what "arr" is here. Is there a better name that we could use?
Also would help if all these functions had summary comments.

Reply:
Very good call. I'll get that cleaned up.